### PR TITLE
Add ember to travis and restructured directories.

### PR DESCRIPTION
**Pull request by matthewphilyaw**
First stab at sorting out ember and travis ci.

I'm adopting this structure:

```
- repo
  | - client (ember)
  | - server (phoenix)
  | - utils (tmux script)
```

### CODE REVIEW

**Comment by matthewphilyaw** [comment]
@copenhas I think this gets us most of the way there, it builds both projects so far and for now coveralls on the phoenix project. 

Anyway the thing I noticed in the build output that was strange is [this](https://travis-ci.org/voluspa/yggdrasil/builds/103187926#L1495-L1507) after calling npm test. However it seems to be related to https://github.com/switchfly/ember-cli-mocha/issues/93, which leads to https://github.com/switchfly/ember-cli-mocha/issues/92 which I believe maybe related to our issue.

Also may need your help with some make files, I just don't think in terms of make and there are several places where we may want to run make commands on not cd into directories like I'm doing. 

A few things to note about the travis ci build
- pay attention to the nvm usage, the default install is actually node .10 something so this is needed
- the export path is for phantomjs 2.0, this was copied from the .travis-ci build in ./client and basically travis ci defaults to phantomjs 1.9.8 on default path, but is available at that path specified so the export is to put it first.
- had to handle the install command and override default behaviour, as mix deps.get needed to be run in ./server.
- most of the node related stuff was from the ./client .travis-ci config. 

Also I know you did coveralls for vilive, could you possibly help me on this? I looked at it briefly but still not sure about the idioms yet.

**Comment by copenhas** [comment]
Oh weird, are we using mocha? I thought the default test library was QUnit. I'm find with using QUnit by the way just to reduce any friction when using Ember.

For Vilive, it looks like I [did some custom stuff](https://github.com/voluspa/vilive/blob/master/Makefile#L18-L28). I'm wondering if ember-cli might have something better as an add-on, but I didn't notice anything right away from a quick search.

Anyway ember-cli actually uses istanbul similar to what I was doing. Looks like they have a [command to run the tests through instanbul](https://github.com/ember-cli/ember-cli/blob/master/package.json#L17). That command is used in the [.travis.yml file](https://github.com/ember-cli/ember-cli/blob/master/.travis.yml#L12) and then later to [post the report to coveralls](https://github.com/ember-cli/ember-cli/blob/master/.travis.yml#L27-L29). Might be easier to try and model it after ember-cli.

**Comment by copenhas** [comment]
I can help you out with basic makefile no problem. It'll simplify the `.travis.yml` file and using the project if we reduce the series of commands for each step to a single make target: `make install`, `make test`, etc. Could maybe modify the tmux script and allow the user to pass in the editor they want or if they want an editor tab through make.

**Comment by matthewphilyaw** [comment]
Ah, no sorry the mocha link led me to the qunit link, seem to be related as the same guy made changes in both which seemed both be about this error. I see that I didn't link the qunit one https://github.com/ember-cli/ember-cli-qunit/pull/93 whoops.

Ok pretty sure it was qunit, getting confused ember-cli's project is mocha, but the ember app it generates uses qunit.

**Comment by matthewphilyaw** [comment]
Cool, yeah I was thinking that maybe we could have like `make install` then like `make test server` and `make test client` because we may be able to leverage parallel builds in travis so being able to split them out may be worth while.

Travis is pretty cool, the more I dig into the more I'm like this is nifty especially offering the free services like they do for open source projects.

File: client/app/app.js [hunk 1]
```diff
@@ -16,10 +16,12 @@ App = Ember.Application.extend({
 loadInitializers(App, config.modulePrefix);
 
 // make bootstrap close the menu after click on an item.
+// jshint ignore: start
 $(document).on('click','.navbar-collapse.in',function(e) {
```
Comment (matthewphilyaw) [inline client/app/app.js:20]: This needs to move to the main-nav component, so ignoring the jshint for now, but will move this soon.

**Comment by matthewphilyaw** [comment]
@copenhas ember tests are sorted now. The next piece is coverage. I found ember-cli-blanket, which looks good but seems to have an issue.

I will try to mull over the istanbul stuff a little more.

**Comment by matthewphilyaw** [comment]
I'm closing this request in favor of splitting repos will issue another one with those changes.

**Pull request closed without merging**